### PR TITLE
#3380 Use appropriate terms for mouse/gamepad/touch

### PR DESCRIPTION
**Pull request by burningserenity**
Show correct buttons for gamepads on item help; change the text for virtual gamepads ever so slightly; refactor logic with choosing how to display misc item text.

Currently, w/ a Dual Shock type controller, the words "triangle" and "square" are written out to represent the buttons, rather than using triangle and square characters to represent them. Using the actual characters causes the game to segfault, leaving this message:
```
ERROR: Missing font kerning: fonts\12-25.bin
ERROR: Missing file: fonts\12-25.pcx
ERROR: Error loading font: fonts\12-25.clx
[1]    16390 segmentation fault  ./build/devilutionx --verbose
```

Looking into this issue.

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -1943,6 +1943,41 @@ void Player::RestorePartialLife()
 	_pHPBase = std::min(_pHPBase + l, _pMaxHPBase);
 }
 
+void Player::RegenLife()
```
Comment (StephenCWills) [inline Source/player.cpp:1946]: FYI, it looks like your life/mana regen code slipped into this PR.

**Comment by AJenbo** [comment]
I'm not sure what you are doing to cause it to have 89 commits. It looks like you have both kept your previous work, possibly reverted it and then rebased upstream on top of your reverted history.

Make sure you start a new feature by first branching the latest upstream master.

File: Source/player.cpp (hunk 1 above)
Comment (burningserenity) [reply Source/player.cpp]: It sure did

**Comment by StephenCWills** [comment]
> Currently, w/ a Dual Shock type controller, the words "triangle" and "square" are written out to represent the buttons, rather than using triangle and square characters to represent them. Using the actual characters causes the game to segfault, leaving this message:
> 
> ```
> ERROR: Missing font kerning: fonts\12-25.bin
> ERROR: Missing file: fonts\12-25.pcx
> ERROR: Error loading font: fonts\12-25.clx
> [1]    16390 segmentation fault  ./build/devilutionx --verbose
> ```
> 
> Looking into this issue.

Maybe I can save you some time. The square and triangle characters you're using likely come from the "Geometric Shapes" Unicode block: `0x25A0`-`0x25CF`. The `12-25.*` files in your log output refers to the font size 12 and the set of characters that begin with `0x25`. The issue is that these files don't exist either in `devilutionx.mpq` or in the extended `fonts.mpq`.

https://github.com/diasurgical/devilutionX/tree/master/Packaging/resources/assets/fonts
https://github.com/diasurgical/devilutionx-assets/tree/main/assets/fonts

In order to use these characters, you'd have to provide a rendering for them in the style of the game's font.

**Comment by AJenbo** [comment]
@burningserenity another option is to provide custom controller symbols in the private use range: U+E000..U+F8FF

This would allow you to render them in any given style unrelated to the generic Geometric Shapes. For example:
https://esquiredevmark.itch.io/generic-gamepad-font

**Pull request closed without merging**